onnx_caffe2/backend.py

```diff
@@ -305,6 +306,68 @@ def _create_optimized_rnn(cls, n):
             else 'linear')
         return op
 
+    @classmethod
+    def _create_slice(cls, n):
+        op = cls._common_onnx_node_to_caffe2_op(n)
+        data, axes, orig_starts, orig_ends = op.input
+        ops = []
+
+        data_shape = dummy_name()
+        ops.append(core.CreateOperator(
+            'Shape',
+            [data],
+            [data_shape]
+        ))
+
+        tmp_starts = dummy_name()
+        starts = dummy_name()
+        ops.extend([
+            core.CreateOperator(
+                'ConstantFill',
+                [data_shape],
+                [tmp_starts],
+                value=0,
+            ),
+            core.CreateOperator(
+                'ScatterAssign',
+                [tmp_starts, axes, orig_starts],
+                [tmp_starts],
+            ),
+            core.CreateOperator(
+                'Cast',
+                [tmp_starts],
+                [starts],
+                to=caffe2_pb2.TensorProto.INT32,
+            ),
+        ])
+
+        tmp_ends = dummy_name()
+        ends = dummy_name()
+        ops.extend([
+            core.CreateOperator(
+                'ConstantFill',
+                [data_shape],
+                [tmp_ends],
+                value=-1,
+            ),
+            core.CreateOperator(
+                'ScatterAssign',
+                [tmp_ends, axes, orig_ends],
```

Comment:
Same here. Otherwise, the types are different.

Reply:
The type of tmp_ends and orig_ends have to be the same (enforced in ScatterAssign). The type of axes and orig_ends could be different, because they are different things (the upperbound of axes is how many dimension a tensor has, while for orig_ends its the individual dimension (could be very big, that's why we need int64)).